File: Java/Program/Animal.java

class Animal {
    protected String gender;
    protected String diet;

    public Animal(String gender, String diet) {
        this.gender = gender;
        this.diet = diet;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public String getDiet() {
        return diet;
    }

    public void setDiet(String diet) {
        this.diet = diet;
    }

    public void display() {
        System.out.println("ANIMAL");
        displayInfo();
    }

    public void displayInfo() {
        System.out.println("> Gender: " + gender);
        System.out.println("> Diet: " + diet);
    }
}

File: CPP/Program/Animal.cpp
#pragma once
#include <iostream>
#include <string>
using namespace std;

class Animal {
protected:
    string gender;
    string diet;

public:
    Animal(const string &g, const string &d) : gender(g), diet(d) {}

    void setGender(const string &g) { gender = g; }
    string getGender() const { return gender; }

    void setDiet(const string &d) { diet = d; }
    string getDiet() const { return diet; }

    virtual void display() const {
        cout << "ANIMAL\n";
        displayInfo();
    }

    virtual void displayInfo() const {
        cout << "> Gender: " << gender << endl;
        cout << "> Diet: " << diet << endl;
    }

    virtual ~Animal() {}
};
